### ISSUE TITLE
C# - CodeLens - Opening file causes code to jump down after "x references" loads

### ISSUE THREAD

**Issue by welkie** [Microsoft/vscode#52054]
What I'm describing is what happens on the screen when you open a source code file that includes reference counts above the first line of its definition or declaration. The moment you open the file, the "x references" line isn't present at all. Then, it will appear above each method and it will say "- references" as VS Code (and, I presume, some extensions) work to determine the references. A short time later, it will change to show you the reference count, for example "2 references", and if you click on this, you get the informative subwindow.

I love this feature, but I tend to work by jumping between multiple source code files often, especially if I only have one monitor at the time. I find it jarring to see my source code jump around as this information loads. Or, I have to wait until the information loads before I start to examine the opened source code file. I also find the idea of having to wait a short time between files annoying.

I would love it if this could be changed so that space on the screen would be allocated as the file is opened for this information to appear later, that way there would be no jarring experience of seeing the code jump down.

If I didn't accurately describe this phenomenon, I can provide a short GIF later.

**Comment by welkie** [comment]
After doing more reading in the docs I've learned that this is the [CodeLens](https://code.visualstudio.com/docs/languages/csharp#_codelens) feature. I learned that I can disable it (as well as enable it on more things like counting interface implementations, very cool) but I don't want to disable it. I just want the editor to be stable once source code files are opened while using this feature. I see this effects more languages than just C# because the feature is available for other languages like TypeScript.

**Comment by Ho-Wan** [comment]
I found the jumping code quite annoying also when loading C# files. There is an extension called "toggle codelens" but it does not work properly with if you have comments in the settings.json file. Therefore I made my own extension that lets me toggle codeLens, but can also be used to toggle other VSCode settings.
[https://marketplace.visualstudio.com/items?itemName=Ho-Wan.setting-toggle](https://marketplace.visualstudio.com/items?itemName=Ho-Wan.setting-toggle)
